plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/PodLogsJob.java

```diff
@@ -189,6 +189,36 @@ protected void runOCBinary(MultiStatus multiStatus) {
 		}
 	}
 
+	/**
+	 * This helper method try not to make a fuss out of closing the input stream internally.
+	 * All other failures should not be hidden.
+	 *  
+	 * @param logs
+	 * @param data
+	 * @return
+	 * @throws IOException
+	 */
+	private int readSafely(InputStream logs, byte[] data) throws IOException {
+		try {
+			if(logs.available() < 0) {
+				return -1;
+			}
+		} catch (IOException e) {
+			//InputStream.available() may throw exception if the stream is closed externally.
+			return -1;
+		}
+		try {
+			return logs.read(data);
+		} catch (IOException e) {
+			if("Stream closed".equals(e.getMessage())) {
```

Comment:
This is thrown by BufferedInputStream. It would be nice to have in InputStream an interfce method isClosed() but I could not find it. The best approximation is the check (InputStream.available() < 0) but if we have BufferedInputStream, it throws that "Stream closed" in that method too instead of returning -1. So, I have decided that these too checks together may be reliable enough.